src/recap.js

```diff
@@ -169,18 +170,54 @@ function Recap() {
           processData: false,
           contentType: false,
           data: formData,
-          success: function(data, textStatus, xhr){
+          success: function (data, textStatus, xhr) {
             console.info(`RECAP: Successfully uploaded PDF: '${textStatus}' ` +
-                        `with processing queue id of ${data['id']}`);
+              `with processing queue id of ${data['id']}`);
             cb(data || null);
           },
-          error: function(xhr, textStatus, errorThrown){
+          error: function (xhr, textStatus, errorThrown) {
             console.error(`RECAP: Ajax error uploading PDF. Status: ${textStatus}.` +
-                          `Error: ${errorThrown}`);
+              `Error: ${errorThrown}`);
           }
         });
 
       });
+    },
+    // Upload a zip file to the RECAP server, calling the cb with ok flag
+    uploadZipFile: async function (pacer_court, pacer_case_id, nonce, cb) {
+      console.info(`RECAP: Attempting PDF upload to RECAP Archive with details: ` +
+        `pacer_court: ${pacer_court}, pacer_case_id: ${pacer_case_id}`)
+
+      // async call for the blob
+      const bytes = await getItemsFromStorage([nonce, 'docId'])
+      console.log("RECAP: Retrieving blob from storage")
+      const ab = new Uint8Array(bytes[nonce])
+      const blob = new Blob([ab])
+
+      let formData = new FormData();
+      formData.append('court', PACER.convertToCourtListenerCourt(pacer_court));
+      pacer_case_id && formData.append('pacer_case_id', pacer_case_id);
+      const docId = bytes['docId']
+      formData.append('pacer_doc_id', docId);
+      formData.append('upload_type', UPLOAD_TYPES['ZIP']);
+      formData.append('debug', DEBUG);
+      formData.append('filepath_local', blob);
+
+      const fetchOptions = {
+        method: 'POST',
+        body: formData,
+        headers: {
+          'Authorization': `Token ${RECAP_TOKEN}`
+        }
+      }
```

Comment:
Presumably we'll be using this kind of a pattern more and more as we move away from jQuery. Is this worth abstracting out? Seems boilerplate-ish.

Reply:
only thing that would be removed and possibly reused is the authorization header, but even if I were to abstract that into a global fetch option, that would also apply to calls made to the pacer sites, so I'm not sure we want that. For a post request, you need to pass in the POST flag and the payload to be uploaded.